### ISSUE TITLE
Custom assets support for in-game editor

### ISSUE THREAD

**Issue by ds-sloth** [Wohlstand/TheXTech#327]
This issue combines multiple suggestions from @LoveBodhi about the in-game editor's handling of custom assets:

[ ] Should be possible for music names to follow `music.ini`
[ ] Include option to remove categorizations from SMBX64 (for blocks, BGOs, tiles) and organize according to asset ID instead.
[ ] Include extended IDs for paths and levels.

This may be missing some of your suggestions, and others may have feedback in this regard too. Please update this issue with that information whenever possible. Thanks!

**Comment by LoveBodhi** [comment]
Right, this is for anyone who building levels with a non-SMBX asset.

**Comment by LoveBodhi** [comment]
The other feature is, it should use image-size based blocks instead of hardcoded / fixed size in the new editor.

**Comment by ds-sloth** [comment]
I'm working on it! The "Cave" section of this block selector was automatically generated based on creator-specifiable information about each block.

![Scr_2022-10-08_22-29-27_(1)](https://user-images.githubusercontent.com/72112344/194734769-3a06fcb6-af53-43c8-ba92-38eee9a17c58.png)

The format for this information includes the block's width, height, "special" variation (unused in this example), slope rate (including for blocks that are next to a slope block), and the set of adjacent blocks.

A part of this example:
```
[s4-cave]
section = 4
name = "Cave"
blk246-adjacent = "632"
blk322-adjacent = "8963214"
blk322-slope = 4
blk322-width = 4
```

The format may change -- if you have comments on this I'd be interested to hear them.

The "adjacent" block set values are in numpad format and are used to make a pleasing arrangement with minimal effort from the asset creator.

**Comment by LoveBodhi** [comment]
There should be a customizable display icon!!

**Comment by ds-sloth** [comment]
Here's an automatically generated layout for extra small-screen devices, and the file that it was automatically generated from. ([editor.ini](https://github.com/Wohlstand/TheXTech/files/9743026/editor.txt)) How does the file format look to you, @LoveBodhi and @Wohlstand?

The current options for blocks are:
* slope (0, 1, 2, or 4)
* width/height (0-7)
* group (0-15) (since some tilesets have more than one subgroup)
* the list of adjacent tiles, in numpad format (so, "124" means that there are tiles to the bottom-left "1", the bottom "2", and the left "4").

![Scr_2022-10-09_20-19-29](https://user-images.githubusercontent.com/72112344/194786139-77cfb8a6-58c2-452f-925e-5d45668eed54.gif)

**Comment by LoveBodhi** [comment]
Feels like acceptable.

**Comment by ds-sloth** [comment]
The most important feature here (customizable editor categories) is present already, so I'm moving this to the next milestone.

**Comment by Shadowblitz16** [comment]
You guys should just do a scrollable list of names for objects with a 32x32 icon.

Items like blocks could be autotile by default and optionally expanded into a grid of the sub types if it supports it.

The issue here is you guys need to optimize for screen space like gba.

And since npcs are not 8x8 tile aligned like in the original games you can't do a grid optiminally.

You should also make the properties a scrollable list too.

Maybe properties could have a ... button which brings up a widget to edit the value.
